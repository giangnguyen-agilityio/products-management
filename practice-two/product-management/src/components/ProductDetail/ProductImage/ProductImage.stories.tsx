import type { Meta, StoryObj } from '@storybook/react'
import ProductImage from './index'

const meta = {
  title: 'Practice Two/ProductImage Component',
  component: ProductImage,
  decorators: [
    Story => (
      <div
        style={{
          padding: '10px',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof ProductImage>

export default meta
type Story = StoryObj<typeof meta>

export const ProductWithImage: Story = {
  args: {
    src: 'https://assets.nicepagecdn.com/d2cc3eaa/3159880/images/yttyy.jpg',
  },
}

export const ProductWithoutImage: Story = {
  args: {
    src: null,
  },
}
