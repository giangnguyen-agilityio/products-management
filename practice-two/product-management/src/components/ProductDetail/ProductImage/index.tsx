// Libraries
import React, { memo } from 'react'
import { Box, Image } from '@chakra-ui/react'

// Assets
import imageNotAvailable from '@assets/images/Image_not_available.webp'

interface ProductImageProps {
  src: string | null
}

const ProductImage: React.FC<ProductImageProps> = ({ src }) => {
  const imageUrl = src || imageNotAvailable

  return (
    <Box
      padding={2}
      border="2px solid"
      borderColor="secondary"
      borderRadius="large"
    >
      <Image
        className="product-image"
        src={imageUrl}
        alt="The product image"
        height={{ base: '200px', sm: '250px', md: '300px', lg: '350px' }}
        objectFit="contain"
        margin="0 auto"
      />
    </Box>
  )
}

export default memo(ProductImage)
