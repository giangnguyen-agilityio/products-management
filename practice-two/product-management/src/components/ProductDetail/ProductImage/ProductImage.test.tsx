import { render } from '@testing-library/react'
import ProductImage from './'

jest.mock('@assets/images/Image_not_available.webp', () => ({
  default: 'imageNotAvailable',
}))

describe('ProductImage Component', () => {
  it('renders with a valid source', () => {
    const { getByAltText } = render(<ProductImage src="valid-image-url.jpg" />)
    const imageElement = getByAltText('The product image')
    expect(imageElement).toBeInTheDocument()
    expect(imageElement).toHaveAttribute('src', 'valid-image-url.jpg')
  })

  it('renders with the default image when source is null', () => {
    const { getByAltText } = render(<ProductImage src={null} />)
    const imageElement = getByAltText('The product image')
    expect(imageElement).toBeInTheDocument()
    expect(imageElement).not.toHaveAttribute(
      'src',
      '@assets/images/Image_not_available.webp'
    )
  })

  it('renders with the default image when source is empty', () => {
    const { getByAltText } = render(<ProductImage src="" />)
    const imageElement = getByAltText('The product image')
    expect(imageElement).toBeInTheDocument()
    expect(imageElement).not.toHaveAttribute(
      'src',
      '@assets/images/Image_not_available.webp'
    )
  })

  it('renders with the correct height based on screen size', () => {
    const { container } = render(<ProductImage src="valid-image-url.jpg" />)
    expect(container).toMatchSnapshot()
  })
})
