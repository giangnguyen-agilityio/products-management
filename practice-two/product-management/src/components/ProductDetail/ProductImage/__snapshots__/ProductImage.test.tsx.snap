// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Component renders with the correct height based on screen size 1`] = `
<div>
  <div
    class="css-1al0hrz"
  >
    <img
      alt="The product image"
      class="product-image css-1wuytdk"
      src="valid-image-url.jpg"
    />
  </div>
</div>
`;
